services:
  web:
    build:
      context: ./django
      args:
        - VOLUME=${DJANGO_WORKDIR:-/home/Auth}
        - LAUNCH=${DJANGO_LAUNCH_REPO:-/run/launch}
    container_name: django_auth
    command: python manage.py runserver 0.0.0.0:8000 
    ports:
      - "8001:8000"
    volumes:
      - ../../Auth:${DJANGO_WORKDIR:-/home/Auth}
    environment:
      - POSTGRES_USER=${SQL_USER:-default_us}
      - POSTGRES_PASSWORD=${SQL_PASSWORD:-default_ps}
      - POSTGRES_DB=${SQL_DATABASE:-default_db}
    env_file:
      - .env
    depends_on:
      db:
        condition: "service_healthy"
    networks:
      - backend
    restart: always

  db:
    container_name: postgreSQL_Auth
    build:
      context: ./postgresql
      args:
        - POSTGRES_USER=${SQL_USER:-default_us}
        - POSTGRES_PASSWORD=${SQL_PASSWORD:-default_ps}
        - POSTGRES_DB=${SQL_DATABASE:-default_db}
    volumes:
      - /home/eslamber/data/Auth:${PSQL_DATA_REPO}
    env_file:
      - .env
    ports:
      - "5431:5432"
    networks:
      - backend
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $USER"]
      start_period: 5s
      retries: 10
      interval: 15s
      timeout: 10s
    restart: on-failure

networks:
  backend:
    external: true